lib/broadway/producer.ex

```diff
@@ -55,4 +70,27 @@ defmodule Broadway.Producer do
       :ok
     end
   end
+
+  defp transform_events(events, transformer) do
+    Enum.map(events, &transform(&1, transformer))
+  end
+
+  defp transform(event, nil) do
+    validate_message(event)
+  end
+
+  defp transform(event, {m, f, opts}) do
+    message = apply(m, f, [event, opts])
+    validate_message(message)
+  end
+
+  defp validate_message(%Message{} = message) do
+    message
+  end
+
+  defp validate_message(_message) do
+    raise "The produced message is invalid. All messages must be a %Broadway.Message{} " <>
```

Comment:
Error messages start with lowercase and we can tell them they can read the `Broadway` docs for more information (or `Broadway.Producer` docs if we decide to move it there).